analyzers/src/SonarAnalyzer.CSharp/SymbolicExecution/Roslyn/ObjectsShouldNotBeDisposedMoreThanOnce.cs

```diff
@@ -25,5 +25,33 @@ public sealed class ObjectsShouldNotBeDisposedMoreThanOnce : ObjectsShouldNotBeD
     public static readonly DiagnosticDescriptor S3966 = DescriptorFactory.Create(DiagnosticId, MessageFormat);
     protected override DiagnosticDescriptor Rule => S3966;
 
-    public override bool ShouldExecute() => false;
+    protected override ILanguageFacade Language => CSharpFacade.Instance;
+
+    public override bool ShouldExecute()
+    {
+        var walker = new Walker();
+        walker.SafeVisit(Node);
+        return walker.Result;
+    }
+
+    private sealed class Walker : SafeCSharpSyntaxWalker
+    {
+        public bool Result { get; private set; }
+
+        public override void Visit(SyntaxNode node)
+        {
+            if (!Result)
+            {
+                base.Visit(node);
+            }
+        }
+
+        public override void VisitInvocationExpression(InvocationExpressionSyntax node)
+        {
+            if (node.GetIdentifier().GetValueOrDefault() is { } methodName && Array.Exists(DisposeMethods, x => x.Equals(methodName.ValueText)))
+            {
+                Result = true;
+            }
```

Comment:
Can you please give me an example with lamba for this case?

Reply:
But in this case, we should still return `false`, because the lambda will be analyzed separately
```
VisitedInvocation(innerSyntaxTreeThatIsNowNotVisited => { d.Dispose(); d.Dispose(); }
```

Reply:
Ok I got it! Thanks!